addons/udes_stock_refactoring/models/stock_picking.py

```diff
@@ -0,0 +1,176 @@
+            # Create a new picking.
+            values = {
+                "picking_type_id": picking_type.id,
+                "location_id": src_loc.id,
+                "location_dest_id": dest_loc.id,
+                "group_id": group.id,
+            }
+            values.update(kwargs)
+
+            picking = self.create(values)
+
+        else:
+            # Avoid misleading values for extra fields.
+            # If any of the fields in kwargs is set and its value is different
+            # than the new one, set the field value to False to avoid misleading
+            # values.
+            # For instance, picking.origin is 'ASN001' and kwargs contains
+            # origin with value 'ASN002', picking.origin is set to False.
+            for field, value in kwargs.items():
+                current_value = getattr(picking, field, None)
+                if isinstance(current_value, models.BaseModel):
+                    current_value = current_value.id
+                if current_value and current_value != value:
+                    setattr(picking, field, False)
+
+        moves.write({"group_id": group.id, "picking_id": picking.id})
+
+        move_lines = moves.mapped("move_line_ids")
+        if move_lines:
+            move_lines.write({"picking_id": picking.id})
+            # After moving move lines check entire packages again just in case
+            # some of the move lines are completing packages
+            if picking.state != "done":
+                picking._check_entire_pack()
+
+        return picking
+
+    def _get_package_search_domain(self, package):
+        """Generate the domain for searching pickings of a package"""
+        return [
+            "|",
+            ("move_line_ids.package_id", "child_of", package.id),
+            ("move_line_ids.result_package_id", "child_of", package.id),
+        ]
+
+    def action_assign(self):
+        """Override action_assign to unlink empty pickings if needed"""
+        res = super(StockPicking, self).action_assign()
+        if self:
+            self.unlink_empty()
+        return res
+
+    def action_confirm(self):
+        """Override action_confirm to unlink empty pickings if needed"""
+        res = super(StockPicking, self).action_confirm()
+        if self:
+            self.unlink_empty()
+        return res
+
+    def action_done(self):
+        """Override action_done to unlink empty pickings if needed"""
+        res = super(StockPicking, self).action_done()
+        if self:
+            self.unlink_empty()
+        return res
+
+    def unlink_empty(self):
+        """
+        Delete pickings in self that are empty, locked and marked=False.
+        If self is empty, find and delete any picking with the previous criterion.
+        This is to prevent us leaving junk data behind when refactoring
+        """
+        records = self.browse()
+        for to_unlink in self.find_empty_pickings_to_unlink():
```

Comment:
Doesn't seem like that function is doing its job if we have to check if there empty. does it need a name change?

Reply:
True, have renamed this to ```get_unmarked_pickings``` as that is all it does, it doesn't do any explicit checks to make sure the pickings it finds are empty.